pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -261,8 +262,8 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
 
     // Calculate routing table for the query
     long routingStartTimeNs = System.nanoTime();
-    Map<String, List<String>> offlineRoutingTable = null;
-    Map<String, List<String>> realtimeRoutingTable = null;
+    Map<ServerInstance, List<String>> offlineRoutingTable = null;
```

Comment:
IIRC this used to be a map of ServerInstance to segments, and we changed it to be String to segments. Not sure what prompted the change. Perhaps @Jackie-Jiang may recall (memory usage?) DNS resolution? (although I see that it is done once in the static map now).